Dwolla.Client/HttpServices/DwollaHttpService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Dwolla.Client.HttpRequestServices;
+using Dwolla.Client.HttpServices.Architecture;
+using Dwolla.Client.Models;
+using Dwolla.Client.Models.Responses;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Dwolla.Client.HttpServices
+{
+	public class DwollaHttpService
+	{
+		private IDwollaClient _client;
+		public IDwollaClient Client
+		{
+			get => _client ?? DwollaConfiguration.Client;
+			set => _client = value;
+		}		
+		
+		private static TokenResponse _cachedToken;
+
+		private Task<string> getAccessTokenAsync;	
```

Comment:
This doesn't work the way I think you're thinking. You're looking for `Action<Task<string>>`. `Task<string>` just represents the result of async operation. Once you have an `Action` you can do `await _getAccessTokenAsync()`.